pilot/pkg/config/kube/crd/client.go

```diff
@@ -70,20 +60,81 @@ type IstioObjectList interface {
 
 // Client is a basic REST client for CRDs implementing config store
 type Client struct {
+	// Map of apiVersion to restClient.
+	clientset map[string]*restClient
+
+	// domainSuffix for the config metadata
+	domainSuffix string
+}
+
+type restClient struct {
+	apiVersion *schema.GroupVersion
```

Comment:
Can you store the `apiversion` by value? Using a pointer doesn't buy us much here and makes the code read less cleanly below re: dereferences.

Reply:
Done